WalletWasabi/Services/WalletService.cs

```diff
@@ -260,10 +253,18 @@ public async Task InitializeAsync(CancellationToken cancel)
 			await RuntimeParams.LoadAsync();
 
 			using (HandleFiltersLock.Lock())
-			using (WalletBlocksLock.Lock())
 			{
+				// Go through the keymanager's index.
+				var bestKeyManagerHeight = KeyManager.GetBestHeight();
+				foreach (var blockstate in KeyManager.GetTransactionIndex())
+				{
+					var block = await GetOrDownloadBlockAsync(blockstate.BlockHash, cancel);
+
+					ProcessBlock(blockstate.BlockHeight, block, blockstate.TransactionIndices);
+				}
+
 				// Go through the filters and que to download the matches.
-				foreach (FilterModel filterModel in Synchronizer.GetFilters().Where(x => !(x.Filter is null) && !WalletBlocks.ContainsValue(x.BlockHash))) // Filter can be null if there is no bech32 tx.
+				foreach (FilterModel filterModel in Synchronizer.GetFilters().Where(x => !(x.Filter is null) && x.BlockHeight >= bestKeyManagerHeight)) // Filter can be null if there is no bech32 tx.
```

Comment:
If I understand correctly we are interested on the filters that we don't have. Is it correct to ask for `>=`?
 

```suggestion
				foreach (FilterModel filterModel in Synchronizer.GetFilters().Where(x => !(x.Filter is null) && x.BlockHeight > bestKeyManagerHeight)) // Filter can be null if there is no bech32 tx.
```

Reply:
Aaah, I don't have permission to accept your commit. I guess because David started the PR. I'll fix it myself.